test/unit/browser/pages/vscode.test.ts

```diff
@@ -175,4 +177,151 @@ describe("vscode", () => {
       localStorage.removeItem("colorThemeData")
     })
   })
+  describe("getLoader", () => {
+    it("should throw an error if options.csStaticBase is undefined or an empty string in context", () => {
+      let options:
+        | {
+            base: string
+            csStaticBase: string
+            logLevel: number
+          }
+        | undefined = {
+        base: "/",
+        csStaticBase: "",
+        logLevel: 1,
+      }
+      const nlsConfig = {
+        first: "Jane",
+        last: "Doe",
+        locale: "en",
+        availableLanguages: {},
+      }
+      const errorMsgPrefix = "[vscode]"
+      const errorMessage = `${errorMsgPrefix} Could not get loader. options or options.csStaticBase is undefined or missing.`
+      expect(() => {
+        if (!options?.csStaticBase) {
+          throw new Error(errorMessage)
+        }
+
+        getLoader({
+          origin: "localhost",
+          nlsConfig: nlsConfig,
+          options,
+        })
+      }).toThrowError(errorMessage)
+      expect(() => {
+        options = undefined
+        if (!options) {
+          throw new Error(errorMessage)
+        }
+        getLoader({
+          origin: "localhost",
+          nlsConfig: nlsConfig,
+          options,
+        })
+      }).toThrowError(errorMessage)
+    })
+    it("should throw an error if nlsConfig is undefined", () => {
+      const options = {
+        base: "/",
+        csStaticBase: "/",
+        logLevel: 1,
+      }
+      const errorMsgPrefix = "[vscode]"
+      const errorMessage = `${errorMsgPrefix} Could not get loader. nlsConfig is undefined.`
+      expect(() => {
+        let nlsConfig = undefined
+
+        if (!nlsConfig) {
+          throw new Error(errorMessage)
+        }
+
+        nlsConfig = {
+          locale: "en",
+          availableLanguages: {},
+        }
+
+        getLoader({
+          nlsConfig,
+          options,
+          origin: "localthost",
+        })
+      }).toThrowError(errorMessage)
+    })
+    it("should return a loader object", () => {
+      const options = {
+        base: "/",
+        csStaticBase: "/",
+        logLevel: 1,
+      }
+      const nlsConfig = {
+        first: "Jane",
+        last: "Doe",
+        locale: "en",
+        availableLanguages: {},
+      }
+      const loader = getLoader({
+        origin: "localhost",
+        nlsConfig: nlsConfig,
+        options,
+      })
+
+      expect(loader).toStrictEqual({
+        baseUrl: "localhost//lib/vscode/out",
+        paths: {
```

Comment:
Testing the paths like this might not catch anything since we'd be copying any mistakes we already have.

The ideal might be to:
1. Ensure that each path resolves to a real Node module.
2. Check VS Code's workbench.html and make sure we have the same paths we do (maybe could do this later).

Reply:
Okay! I'll take a look adding here. Worst case, I'll create a follow-up issue and improving there

Reply:
Added this as a follow-up item: https://github.com/cdr/code-server/issues/3781